core/src/main/scala/org/apache/spark/ui/WebUI.scala

```diff
@@ -126,7 +126,7 @@ private[spark] abstract class WebUI(
   def bind(): Unit = {
     assert(serverInfo.isEmpty, s"Attempted to bind $className more than once!")
     try {
-      val host = Option(conf.getenv("SPARK_LOCAL_IP")).getOrElse("0.0.0.0")
+      val host = conf.get(DRIVER_BIND_ADDRESS)
```

Comment:
This actually changes behavior, because now you'll always be listening on a specific interface instead of "0.0.0.0".

Try it out: run spark-shell in local mode; it should try to figure out your non-loopback address for the driver, but you should still be able to connect to "127.0.0.1:4040" in your browser. With your changes that will probably stop working.

Reply:
This is a great point, which helped identify another bug where this `host` is entirely ignored. 

reworked it to behave as was intended on the client node.